### PR TITLE
Various bug fixes

### PR DESCRIPTION
**Pull request by panleone**
Solve a couple of bug found during testing:

- Fix a bug in finalizeTransaction
- Fix documentation for the list of unspent notes
- Fix shield address repetition in zip32

### CODE REVIEW

**Comment by panleone** [comment]
Soo I kept testing the full Shield system on MPW and I found another couple of problems: the main one is that internal change address is completely broken and any PIVs sent there are lost forever, so we should switch back to a normal shield address  for changes,

File: src/keys.rs [hunk 1]
```diff
@@ -72,16 +72,20 @@ pub fn generate_next_shielding_payment_address(
...
             .to_diversifiable_full_viewing_key()
             .find_address(diversifier_index);
         if let Some(payment_address) = payment_address {
-            found_addresses += 1;
-            if found_addresses == n_address {
-                let enc_addr = encode_payment_address(&payment_address.1, is_testnet);
-                return serde_wasm_bindgen::to_value(&enc_addr)
-                    .expect("Cannot serialize payment address");
+            let enc_addr = encode_payment_address(&payment_address.1, is_testnet);
+            if !(enc_addr == last_addres) {
```
Comment (Duddino) [inline src/keys.rs:82]:
```suggestion
            if enc_addr != last_addres {
```

File: src/transaction.rs [hunk 2]
```diff
@@ -322,15 +323,33 @@ pub async fn create_transaction_internal(
     block_height: BlockHeight,
     network: Network,
 ) -> Result<JSTransaction, Box<dyn Error>> {
+    let mut amount = amount;
```
Comment (Duddino) [inline src/transaction.rs:326]:
```suggestion
```
Remove this line and add mut in the parameters

**Review by Duddino — CHANGES_REQUESTED** [review]

File: src/keys.rs [hunk 3]
```diff
@@ -72,16 +72,20 @@ pub fn generate_next_shielding_payment_address(
     let extsk = decode_extsk(&enc_extsk, is_testnet);
     let mut found_addresses = 0;
     let mut diversifier_index = DiversifierIndex::new();
+    let mut last_addres: String = "".to_string();
     loop {
```
[the remark this answers is not in the document]
Comment (Duddino) [reply src/keys.rs:75]: Deferred to #30

File: src/keys.rs [hunk 4]
```diff
@@ -72,20 +72,17 @@ pub fn generate_next_shielding_payment_address(
     let extsk = decode_extsk(&enc_extsk, is_testnet);
     let mut found_addresses = 0;
     let mut diversifier_index = DiversifierIndex::new();
-    let mut last_addres: String = "".to_string();
     loop {
-        let payment_address = extsk
+        let bundle_adress = extsk
```
Comment (Duddino) [inline src/keys.rs:76]: typo

**Review by Duddino — APPROVED** [review]
tACK other than the typo

**Pull request merged**